### ISSUE TITLE
Exporting to pbf only populate 99 objects with data

### ISSUE THREAD

**Issue by soli004** [roelderickx/ogr2osm#40]
I'm using MacOSX and have a shape (tried with geopackage as well) with road data. When using this python script to safe to pbf file it does not matter the size of the file, it only populate 99 line objects the rest is empty with not even 'null'. The geometry is there I can see the lines and I can select them but there is no whatsoever values. I tried with 100 000 objects, 1000 objects, 256 objects and it is always only 99 objects with features. I then saved a file with 70 objects and this got complete with data, all lines (roads) had name, highway etc. It seems to be a limit somewhere that I don't understand how to solve. I call this script from the terminal. Any input in this issue for a fix would be mostly welcome.
import logging
import ogr2osm
from datetime import datetime

# Set up the logger for ogr2osm to capture debug logs
ogr2osmlogger = logging.getLogger('ogr2osm')
ogr2osmlogger.setLevel(logging.DEBUG)
ogr2osmlogger.addHandler(logging.StreamHandler())

# Parameters
datasource_parameter = "/Users/username/Downloads/test.shp"

# Create a timestamp for the output file name
current_time = datetime.now().strftime('%Y%m%d-%H%M%S')
output_file = f"/Users/username/Downloads/roads_{current_time}.osm.pbf"

# Create the translation object (assuming you're using the base class)
translation_object = ogr2osm.TranslationBase()

# Create the OGR datasource and open it
datasource = ogr2osm.OgrDatasource(translation_object)
datasource.open_datasource(datasource_parameter)

# Convert the OGR data to OSM format
osmdata = ogr2osm.OsmData(translation_object)
osmdata.process(datasource)

# Write the OSM data to an output file
datawriter = ogr2osm.OsmDataWriter(output_file)
osmdata.output(datawriter)

**Comment by roelderickx** [comment]
The script you posted is fine, at least if you want osm output. If you want pbf output you should swap OsmDataWriter for PbfDataWriter, but temporarily generating osm files is easier for debugging.

Note that JOSM struggles with osm files if they have a pbf extension, it may be applicable to the software you use as well.

I tested with the testshapefile.shp you can find in the test/shapefiles folder of this repository, resulting in an osm file with 522 nodes and 114 ways. It seems to work fine beyond the 99th object.

So there is either something wrong with your input file or something strange happens in the software you're using to view the resulting osm file. Since I was able to convert the testshapefile.shp with your script I suggest you try the same and investigate the result in a text editor. Are all 522 nodes and 114 ways correctly generated?
If yes then open the same file in the software of your preference and see if all nodes and ways are still there.

**Comment by soli004** [comment]
Thank you, yes you are correct, I used QGIS to check the result and it seems to have limitations to display attributes from a pbf file that I did not know about. When saving the pbf file to shape or geopackage and opened in QGIS all the attributes etc was in place. QGIS can open pbf files and save to other formats but displaying pbf attributes is not working properly. I did try the testshapefile and got similar result of 99 out of 114 so that made me suspicious of QGIS handling the pbf. Thanks for the info, I was stuck with this one....
